### PR TITLE
Add apazzy keymap to GMMK Pro v1 ANSI

### PR DESCRIPTION
**Pull request by apazzy**

### CODE REVIEW

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:1]: This file is missing a license header.

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+#include "rgb_matrix_map.h"
+
+#define KC_TASK LCTL(LSFT(KC_ESC))
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:4]: Not used anywhere?

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+enum layers{
+    WIN_BASE,
+    WIN_FN,
+};
+
+enum apazzy_keycodes {
+  KC_WINLCK = SAFE_RANGE,
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:12]:
```suggestion
  KC_WINLCK = QK_USER,
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+void matrix_scan_user(void) {
+    if (mouse_jiggle_mode) {
+        tap_code(KC_MS_UP);
+        tap_code(KC_MS_DOWN);
+        tap_code(KC_MS_LEFT);
+        tap_code(KC_MS_RIGHT);
+        tap_code(KC_MS_WH_UP);
+        tap_code(KC_MS_WH_DOWN);
+    } else {
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:58]:
```suggestion
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+                }
+                return true;
+            }
+            return true;
+        case EMO_SHRUG:
+            if (record->event.pressed) {
+                SEND_STRING("¯\\_(ツ)_/¯");
+            }
+            return false;
+		// process all other keycodes normally
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:100]: This won't work — `SEND_STRING()` only supports ASCII characters.
```suggestion
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_RST2:
+            if (record->event.pressed) {
+                reset_keyboard();
+            }
+            return false;
+		// GUI lock toggle
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:69]: I don't see how this works, but `QK_BOOT` doesn't? Sounds like there's some other issue at play.

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+        case KC_WINLCK:
+            if (record->event.pressed) {
+                keymap_config.no_gui = !keymap_config.no_gui; //toggle status
+            }
+            return false;
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:74]: The [magic](https://docs.qmk.fm/#/keycodes_magic)  keycode `GU_TOGG` should be used instead.
```suggestion
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/readme.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+FN+WIN - Windows key lock toggle
+
+- `KC_WINLCK`   GUI/*WIN*dows key *L*o*CK*
+- `KC_MJST`     *M*ouse *J*iggler *ST*art
+- `KC_MJCX`     *M*ouse *J*iggler *CX*ancel
+
+Not mapped yet but I want to map these at some point.
+- `KC_TASK`     *TASK* Manager (`LCTL(LSFT(KC_ESC)))`)
+- `EMO_SHRUG`   shrug guy: ¯\\_(ツ)_/¯ (not tested)
+
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/readme.md:20]:
```suggestion
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/rgb_matrix_map.h [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (waffle87) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/rgb_matrix_map.h:16]:
```suggestion
#pragma once
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by waffle87** [comment]
Please use and fill out the [template](https://github.com/qmk/qmk_firmware/blob/master/.github/PULL_REQUEST_TEMPLATE.md) that was provided.

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c (hunk 5 above)
Comment (drashna) [reply keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:100]: Would need unicode for this, specifically.

File: keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+        return false;
+    }
+
+    void suspend_power_down_user(void) {
+        rgb_matrix_set_suspend_state(true);
+    }
+
+    void suspend_wakeup_init_user(void) {
+        rgb_matrix_set_suspend_state(false);
+    }
```
Comment (drashna) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/apazzy/keymap.c:160]:
```suggestion
```

You only need this in your config.h to enable sleep:
```c
#define RGB_DISABLE_WHEN_USB_SUSPENDED
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**